### PR TITLE
CFE-2857: Reload persistent classes and fix related memory errors

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: cf-serverd/cf-serverd-functions.c [hunk 1]
```diff
@@ -455,8 +455,8 @@ static void CheckFileChanges(EvalContext *ctx, Policy **policy, GenericAgentConf
             UpdateLastPolicyUpdateTime(ctx);
 
             DetectEnvironment(ctx);
+            GenericAgentDiscoverContext(ctx, config);
             KeepHardClasses(ctx);
```
Comment (nickanderson) [inline cf-serverd/cf-serverd-functions.c]: What does KeepHardClasses do?
It's important that augments is loaded after environment detection and hard classes are set.

Comment (olehermanse) [reply cf-serverd/cf-serverd-functions.c]: Everything `KeepHardClasses` does is already done by `GenericAgentDiscoverContext`. I will try to remove it.

File: cf-execd/cf-execd.c [hunk 2]
```diff
@@ -628,9 +628,7 @@ static bool ScheduleRun(EvalContext *ctx, Policy **policy, GenericAgentConfig *c
         UpdateLastPolicyUpdateTime(ctx);
 
         DetectEnvironment(ctx);
-
-        Log(LOG_LEVEL_INFO, "Re-evaluating augments" );
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply cf-execd/cf-execd.c]: This log message only appeared in cf-execd. It is better that they are the same across all binaries, imo.

The current message for all binaries is verbose:
```
Log(LOG_LEVEL_VERBOSE, "Loading JSON augments from '%s' (input dir '%s', input file '%s'", def_json, config->input_dir, config->input_file);
```

[the remark this answers is not in the document]
Comment (nickanderson) [reply cf-execd/cf-execd.c]: OK good.

**Comment by olehermanse** [comment]
@cf-bottom jenkins with exotics

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1590/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1590/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1591/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1591/

**Comment by olehermanse** [comment]
@vpodzime OK to cherry-pick?

**Pull request merged**